spec/string-class-spec.js

```diff
@@ -0,0 +1,201 @@
+  describe('Test hasVowels', () => {
+    it('should return false if the string has no vowels', () => {
+      expect('N VWLS'.hasVowels()).toBe(false);
+      expect('n vwls'.hasVowels()).toBe(false);
+      expect('@#$^&*%12345'.hasVowels()).toBe(false);
+      expect(' '.hasVowels()).toBe(false);
+    });
+
+    it('should return true if the string has vowels', () => {
+      expect('No vowels'.hasVowels()).toBe(true);
+      expect('NO VOWELS'.hasVowels()).toBe(true);
+      expect('n0 v0wels'.hasVowels()).toBe(true);
+      expect('@#$^&*% 12345 e'.hasVowels()).toBe(true);
+    });
+
+    it('should return a boolean value', () => {
+      expect(typeof ('1234'.hasVowels())).toBe('boolean');
+    });
+
+    it('should return a defined value', () => {
+      expect(typeof ('1234'.hasVowels())).toBeDefined();
+    });
+  });
+
+  describe('Test toUpper', () => {
+    it('should return a string', () => {
+      expect(typeof 'Lowercase'.toUpper()).toBe('string');
+    });
+
+    it('should return the string passed in uppercase', () => {
+      expect('lowercase'.toUpper()).toEqual('LOWERCASE');
+      expect('LowerCase'.toUpper()).toEqual('LOWERCASE');
+      expect('L0werCAs3& letters'.toUpper()).toEqual('L0WERCAS3& LETTERS');
+      expect(''.toUpper()).toEqual('');
+    });
+
+    it('should verify that the inbuilt toUpper string method was not called', () => {
+      spyOn(String.prototype, 'toUpperCase');
+
+      // Trigger the toUpper method
+      const spoof = 'JamesBond';
+      spoof.toUpper();
+
+      // Check whether the native javaScript toUpperCase was called
+      expect(String.prototype.toUpperCase).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('Test toLower', () => {
+    it('should return a string', () => {
+      expect(typeof 'Lowercase'.toLower()).toBe('string');
+    });
+
+    it('should return the string passed in lowercase', () => {
+      expect('LOWERCASE'.toLower()).toEqual('lowercase');
+      expect('LowerCase'.toLower()).toEqual('lowercase');
+      expect('L0werCAs3& letters'.toLower()).toEqual('l0wercas3& letters');
+      expect(''.toLower()).toEqual('');
+    });
+
+    it('should verify that the inbuilt toUpper string method was not called', () => {
+      spyOn(String.prototype, 'toLowerCase');
+      const spoof = 'Ian Flemming';
+      spoof.toLower();
+      expect(String.prototype.toLowerCase).not.toHaveBeenCalled();
+    });
+  });
+
+  describe('Test upperCaseFirst', () => {
+    it('should return a string', () => {
+      expect(typeof 'Uppercase'.ucFirst()).toBe('string');
+    });
+
+    it('should return the input string with its first letter in uppercase', () => {
+      expect('UCFIRST'.ucFirst()).toEqual('UCFIRST');
+      expect('upperCasefirst'.ucFirst()).toEqual('UpperCasefirst');
+      expect('uppercase first'.ucFirst()).toEqual('Uppercase first');
+    });
+
+    it('should return an empty string for an empty string as an input', () => {
+      expect(''.ucFirst()).toEqual('');
```

Comment:
**Code Quality : Nitpick **
Replace return 'invalid' for blank input string with return ' '
```expect(''.ucFirst()).toEqual('');```